var/spack/repos/builtin/packages/py-multiqc/package.py

```diff
@@ -14,28 +14,50 @@ class PyMultiqc(PythonPackage):
     version("1.14", sha256="dcbba405f0c9521ed2bbd7e8f7a9200643047311c9619878b81d167300149362")
     version("1.13", sha256="0564fb0f894e6ca0822a0f860941b3ed2c33dce407395ac0c2103775d45cbfa0")
 
+    # dependency defintions move from setup.py to pyproject.toml as of @1.21:
+
+    # build deps
     depends_on("py-setuptools", type="build")
-    depends_on("py-matplotlib@2.1.1:", type=("build", "run"))
-    depends_on("py-networkx@2.5.1:", type=("build", "run"))
-    depends_on("py-numpy", type=("build", "run"))
+
+    # current run deps
     depends_on("py-click", type=("build", "run"))
-    depends_on("py-coloredlogs", type=("build", "run"))
-    depends_on("py-future@0.14.1:", type=("build", "run"))
+    depends_on("py-humanize", type=("build", "run"), when="@1.18:")
+    depends_on("py-importlib-metadata", type=("build", "run"), when="@1.16:")
     depends_on("py-jinja2@3.0.0:", type=("build", "run"), when="@1.14:")
     depends_on("py-jinja2@2.9:", type=("build", "run"), when="@:1.13")
-    depends_on("py-lzstring", type=("build", "run"))
-    depends_on("py-markdown", type=("build", "run"))
+    depends_on("py-kaleido", type=("build", "run"), when="@1.20:")
+    depends_on("py-markdown", type=("build", "run"), when="@1.3:")
+    depends_on("py-matplotlib@2.1.1:", type=("build", "run"), when="@1.13:")
+    depends_on("py-matplotlib@2.1.1:2", type=("build", "run"), when="@1.7")
+    depends_on("py-matplotlib@:2.1.0", type=("build", "run"), when="@1.5")
+    depends_on("py-matplotlib", type=("build", "run"), when="@:1.3")
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-packaging", type=("build", "run"), when="@1.16:")
+    depends_on("py-requests", type=("build", "run"), when="@1.3:")
+    depends_on("py-pillow@10:", type=("build", "run"), when="@1.20:")
+    depends_on("py-plotly@5.18:", type=("build", "run"), when="@1.21:")
+    depends_on("py-plotly", type=("build", "run"), when="@1.20")
+    depends_on("py-pyyaml@4:", type=("build", "run"), when="@1.13:")
     depends_on("py-pyyaml", type=("build", "run"))
-    depends_on("py-pyyaml@4:", type=("build", "run"))
-    depends_on("py-requests", type=("build", "run"))
-    depends_on("py-rich@10:", type=("build", "run"))
-    depends_on("py-rich-click", type=("build", "run"))
-    depends_on("py-simplejson", type=("build", "run"))
-    depends_on("py-spectra@0.0.10:", type=("build", "run"))
-    depends_on("py-spectra", type=("build", "run"))
+    depends_on("py-pyaml-env", type=("build", "run"), when="@1.18:")
+    depends_on("py-rich@10:", type=("build", "run"), when="@1.13:")
+    depends_on("py-rich-click", type=("build", "run"), when="@1.13:")
+    depends_on("py-coloredlogs", type=("build", "run"), when="@1.13:")
+    depends_on("py-tqdm", type=("build", "run"))
+    depends_on("py-spectra@0.0.10:", type=("build", "run"), when="@1.4:")
+    depends_on("py-spectra", type=("build", "run"), when="@1.18:")
+    depends_on("py-typeguard", type=("build", "run"))
```

Comment:
I'm not seeing `tqdm` or `typeguard` in the `pyproject.toml` for the latest version.

Reply:
Oops, those are actually from `@1.22:` - I must have been viewing the `main` branch at some point by mistake when I was flicking back and forth.

(I haven't updated all the way to the current `@1.22.2` yet as there's a new dependency on `pydantic@2.7.1:` that appears fiddly, so I'll come back to it when there's time...)